pkgs/swift2objc/lib/src/parser/parsers/declaration_parsers/parse_initializer_declaration.dart

```diff
@@ -0,0 +1,69 @@
+import '../../../ast/_core/shared/parameter.dart';
+import '../../../ast/declarations/compounds/members/initializer_declaration.dart';
+import '../../_core/json.dart';
+import '../../_core/parsed_symbolgraph.dart';
+import '../../_core/utils.dart';
+
+InitializerDeclaration parseInitializerDeclaration(
+  Json initializerSymbolJson,
+  ParsedSymbolgraph symbolgraph,
+) {
+  return InitializerDeclaration(
+    id: parseSymbolId(initializerSymbolJson),
+    params: _parseInitializerParams(initializerSymbolJson, symbolgraph),
+    hasObjCAnnotation: symbolHasObjcAnnotation(initializerSymbolJson),
+  );
+}
+
+List<Parameter> _parseInitializerParams(
+  Json initializerSymbolJson,
+  ParsedSymbolgraph symbolgraph,
+) {
+  final fragments = initializerSymbolJson['declarationFragments'];
+
+  final externalParamKind = 'externalParam';
+  final internalParamKind = 'internalParam';
+  final typeIdentifierKind = 'typeIdentifier';
+
+  String? externalParam;
+  String? internalParam;
+  String? typeId;
+
+  final parameters = <Parameter>[];
+
+  for (final fragmentJson in fragments) {
```

Comment:
This loop is pretty complicated. I had to figure out how the params were stored by reading this code 🙃. Can you add a comment describing the structure you're parsing and the rules you're enforcing? Something like this (but your own words): `// Initializer params are stored in a flat array like [externalName, internalName, type, externalName, internalName, type, ...]. The type always comes at the end of the grouping, and there's only ever allowed to be one ex/int name per group. Internal name is optional, external name is required.`

Reply:
I added comments and refactored the loop code to be more explicit about what it's doing